### PR TITLE
feat: add support to PSC ip type

### PR DESCRIPTION
**Pull request by ruyadorno**
This PR adds support to a new `IpAddressTypes.PSC` ip type to support conections to Cloud SQL private service connect.

Ref: https://cloud.google.com/vpc/docs/private-service-connect

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -230,6 +213,35 @@ connection.close();
 connector.close();
 ```
 
+### Specifying Public or Private IP
```
Comment (jackwotherspoon) [inline README.md:216]: We can probably fold the `Using a Private Service Connect (PSC) IP` heading into this one to make things more concise. Maybe renaming it to `Specifying IP Address Type` like python connector PR?: https://github.com/GoogleCloudPlatform/cloud-sql-python-connector/pull/766/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

File: src/ip-addresses.ts [hunk 2]
```diff
@@ -45,27 +47,37 @@ const getPrivateIpAddress = (ipAddresses: IpAddresses) => {
...
+      message: 'Cannot connect to instance, PSC address not found',
+      code: 'ENOPSCSQLADMINIPADDRESS',
     });
   }
+  return ipAddresses.psc;
+};
 
+export function parseIpAddresses(
+  ipResponse: sqladmin_v1beta4.Schema$IpMapping[] | undefined,
+  dnsName: string | null | undefined
```
Comment (jackwotherspoon) [inline src/ip-addresses.ts:62]: if `PSC` is not enabled on an instance, does the node client return `null` for `dnsName`? I know the Go client defaults it to empty string.

File: test/ip-addresses.ts [hunk 3]
```diff
@@ -20,86 +20,132 @@ import {
...
+  ),
   {
     private: '0.0.0.2',
     public: '0.0.0.0',
   },
   'should return a both public and private ips if available'
 );
 
+t.same(
+  parseIpAddresses([], '0.0.0.3'),
```
Comment (jackwotherspoon) [inline test/ip-addresses.ts:116]: can we use `'abcde.12345.us-central1.sql.goog'` as the test `dnsName` variable? This is the format the dnsName takes in an actual response.

**Review by jackwotherspoon — CHANGES_REQUESTED** [review]
Small nits, otherwise looks great!

File: src/ip-addresses.ts (hunk 2 above)
Comment (ruyadorno) [reply src/ip-addresses.ts:62]: Good question, I'm following the type definitions from [`@googleapis/sqladmin`](https://github.com/googleapis/google-api-nodejs-client/blob/main/src/apis/sqladmin/v1beta4.ts#L427) which sounds like the safest approach here.

Comment (jackwotherspoon) [reply src/ip-addresses.ts:62]: Sounds good to me! 👍

**Review by jackwotherspoon — APPROVED** [review]
LGTM ✅

**Pull request merged**